MAINTAINING.md

```diff
@@ -0,0 +1,147 @@
+# RIOT Review Guidelines
+
+The items presented in the following list should be checked for each PR review,
+in this order. This list is however not exhaustive and defines a rather minimal
+baseline on things that should be checked.
+
+Note: Any of the items in this list might be skipped, if **good** reasons are presented.
+
+Also note: This reviewing order is meant to maximize efficiency and minimize overhead, since failing in step n makes steps n+x obsolete. Maintainers should be aware that starting a review with 4. (Code Style) not only wasts their own time, but also the time of the author
+
+## 1. - Review the fundaments
+
+1.1 Does the reasoning for this PR make sense?
+1.2 Do the concepts used by the PR make sense? 
+1.3 Does the PR break with existing concepts?
+1.4 Is the structure of the PR itself valid?
+1.5 Are the testing criteria clear?
```

Comment:
> unnamed: 1.5: does this mean transperency on how to reproduce and est this one PR? e.g., if a special setup is needed, how to set this up? maybe we can concretise this point

I agree with this comment (it might be my own, but I'm not sure). IMHO this point should not mean "Contributor, please provide a test applications or scripted test case" (unless they are really straight forward or prove that the PR actually improves upon the current code base or because the contributor wants to). This could (and should) e.g. interpreteted as that the API (inkl. documentation) is clear enough, that a third party can implement comprehensive tests (ideally black box) on their own. Or in case of a bug fix of course: A description how to reproduce the bug and how the PR fixes that.

And yes, this is a diversion from our current handling of tests.